server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraWithTikaTest.java

```diff
@@ -19,28 +19,33 @@
 
 package org.apache.james;
 
-import java.io.IOException;
-
-import org.junit.ClassRule;
-import org.junit.Rule;
-
-public class CassandraWithTikaTest extends AbstractJamesServerTest {
-
-    @ClassRule
-    public static final DockerCassandraRule cassandra = new DockerCassandraRule();
-    @ClassRule
-    public static final GuiceTikaRule guiceTikaRule = new GuiceTikaRule();
-
-    @Rule
-    public CassandraJmapTestRule cassandraJmap = CassandraJmapTestRule.defaultTestRule();
-
-    @Override
-    protected GuiceJamesServer createJamesServer() throws IOException {
-        return cassandraJmap.jmapServer(guiceTikaRule.getModule(), cassandra.getModule(), DOMAIN_LIST_CONFIGURATION_MODULE);
-    }
-
-    @Override
-    protected void clean() {
-    }
-
+import static org.apache.james.CassandraJamesServerMain.ALL_BUT_JMX_CASSANDRA_MODULE;
+
+import org.apache.james.mailbox.extractor.TextExtractor;
+import org.apache.james.mailbox.store.search.PDFTextExtractor;
+import org.apache.james.modules.TestESMetricReporterModule;
+import org.apache.james.modules.TestJMAPServerModule;
+import org.apache.james.server.core.configuration.Configuration;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.google.inject.util.Modules;
+
+class CassandraWithTikaTest implements JamesServerContract {
+
+    private static final int LIMIT_TO_10_MESSAGES = 10;
+
+    @RegisterExtension
+    static JamesServerExtension testExtension =
+        new JamesServerExtensionBuilder()
+            .extension(CassandraExtension::new)
+            .extension(TikaExtension::new)
+            .extension(EmbeddedElasticSearchExtension::new)
+            .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
+                .combineWith(ALL_BUT_JMX_CASSANDRA_MODULE)
+                .overrideWith(Modules.combine(
+                    binder -> binder.bind(TextExtractor.class).to(PDFTextExtractor.class),
+                    new TestJMAPServerModule(LIMIT_TO_10_MESSAGES),
+                    new TestESMetricReporterModule()))
+                .overrideWith(DOMAIN_LIST_CONFIGURATION_MODULE))
+            .build();
```

Comment:
This builder looks really hard to use to me. Module declaration are also duplicated everywhere.

In order to launch a single test, do I really need to specify the full server details, all the time?

Reply:
"really hard" seems a bit overstated (: 

What are the solutions to make it shorter ?

We can have pre-defined AggregateExtension with Cassandra and ElasticSearch but I'm not sure it's worth the one line gain.
We can remove the `.server(configuration GuiceJamesServer.forConfiguration(configuration))` part and replace it with `.server(server -> server.combine...)`.
For the last 3 lines, I think that's important to decide which modules you want to run and what override you want.
Do you see any other information that we could remove/hide without loosing control over what we test ?

Reply:
At least I think we should provide a constant of default `JamesServerExtension` at somewhere. I experienced the duplications from #1763 and I feel dizzy. Some tests it doesn't require us to know too much about how to build a `JamesServerExtension` (maybe building a specific james server is not what the test targets to or want to draw our attention). The old version `JamesServerTestRule` has a `defaultTestRule()` static method, many tests are using it.